src/configurationProvider.ts

```diff
@@ -88,13 +88,19 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
                 await updateDebugSettings();
             }
 
-            if (Object.keys(config).length === 0) { // No launch.json in current workspace.
-                // check whether it is opened as a folder
-                if (folder !== undefined) {
-                    // for opened with folder, return directly.
-                    return config;
-                }
-                // only rebuild for single file case before the build error issue is resolved.
+            // just create and open the lanuch.json if there is no lanch.json in current folder
+            if (Object.keys(config).length === 0 && folder !== undefined) {
+                return config;
+            }
+
+            // generate config in memory if there is noly single file and no launch.json
+            if (Object.keys(config).length === 0 && !folder) {
+                config.type = "java";
+                config.name = "Java Debug";
+                config.request = "launch";
+            }
+
+            if (config.request === "launch") {
                 try {
                     const buildResult = await vscode.commands.executeCommand(commands.JAVA_BUILD_WORKSPACE, false);
                     console.log(buildResult);

```

Comment:
please remove the console.log() but the logic to check build status with the right kind of error message.